scrapy/downloadermiddlewares/cookies.py

```diff
@@ -51,7 +51,13 @@ def __init__(self, debug: bool = False):
     def from_crawler(cls, crawler: Crawler) -> Self:
         if not crawler.settings.getbool("COOKIES_ENABLED"):
             raise NotConfigured
-        return cls(crawler.settings.getbool("COOKIES_DEBUG"))
+        o = cls(crawler.settings.getbool("COOKIES_DEBUG"))
+        if crawler.settings.getbool("COOKIES_EXPOSE_JARS"):
+            crawler.signals.connect(o.spider_opened, signal=signals.spider_opened)
+        return o
+
+    def spider_opened(self, spider: Spider) -> None:
+        spider.jars = self.jars
```

Comment:
What about using a more descriptive name for the spider attribute? `spider.cookie_jars`?